controller/src/main/java/org/jboss/as/controller/AbstractOperationContext.java

```diff
@@ -541,6 +541,41 @@ public final String getCurrentAddressValue() {
         return pa.getLastElement().getValue();
     }
 
+    @Override
+    public final ModelNode getCurrentOperationName() {
+        assert activeStep != null;
+        ModelNode operation = activeStep.operation;
+
+        assert operation != null;
+        return operation.get(NAME);
+    }
+
+    @Override
+    public final ModelNode getCurrentOperationParameter(final String parameterName) {
+        return getCurrentOperationParameter(parameterName, true);
+    }
+
+    @Override
+    public final ModelNode getCurrentOperationParameter(final String parameterName, boolean nullable) {
+        if (isLegalParameterName(parameterName)) {
+            assert activeStep != null;
+            ModelNode operation = activeStep.operation;
+
+            assert operation != null;
+            if (!operation.has(parameterName) && nullable) {
+                return null;
+            } else {
+                return operation.get(parameterName);
+            }
+        } else {
+            throw new IllegalArgumentException(ControllerLogger.ROOT_LOGGER.invalidParameterName(parameterName));
```

Comment:
ControllerLogger.ROOT_LOGGER.invalidParameterName should return IllegalArgumentException so you can just throw what it returns without wrapping.